netket/graph/lattice.py

```diff
@@ -531,13 +500,119 @@ def position_from_basis_coords(self, basis_coords: CoordT) -> PositionT:
         Return the position of the site with given basis coordinates.
         When passed a rank-2 array where each row is a coordinate vector,
         this method returns an array of the corresponding positions.
-        Throws a KeyError if no site is found for any of the coordinates.
+        Throws an `InvalidSiteError` if no site is found for any of the coordinates.
         """
         ids = self.id_from_basis_coords(basis_coords)
-        if _np.any(ids == None):  # pylint: disable=singleton-comparison
-            raise KeyError(f"No site found at at least one of {basis_coords}")
         return self.positions[ids]
 
+    def to_reciprocal_lattice(self, ks: Array) -> Array:
+        """
+        Converts wave vectors from Cartesian axes to reciprocal lattice vectors.
+
+        Arguments:
+            ks: wave vectors in Cartesian axes. Multidimensional arrays are accepted,
+                the Cartesian coordinates must form the last dimension.
+
+        Returns:
+            The same wave vectors in the reciprocal basis **of the simulation box.**
+            Valid wave vector components in this basis are integers in (periodic BCs)
+            or zero (in open BCs).
+
+        Throws an `InvalidWaveVectorError` if any of the supplied wave vectors
+        are not reciprocal lattice vectors of the simulation box.
+        """
+        # Ensure that ks has at least 2 dimensions
+        ks = _np.asarray(ks)
+        if ks.ndim == 1:
+            ks = ks[_np.newaxis, :]
+
+        result = ks @ self._lattice_dims.T / (2 * pi)
+        # Check that these are integers
+        is_valid = is_approx_int(result)
```

Comment:
I don't think this "is_valid" check is correct. You can define the same lattice by dividing the dimensions of unit_cell and basis_vectors by a constant. This scales down _lattice_dims and changes which wave vectors are valid.

Reply:
Should unit cells always have volume 1 or something like that?

Reply:
It is valid, `result` is the wave vector in units reciprocal lattice vectors. If the unit cell gets smaller, the BZ gets larger, so the equivalent wave vectors get larger.

Reply:
Why does this break then?

```
graph = Lattice(basis_vectors=[[1/4,0],[0,1/4]],extent=[3,3])
graph.space_group_builder(group.planar.D(4)).little_group([pi,pi])
```

Reply:
Okay, I see your point. Lets make sure our default lattices have unit volume then?